backend/src/export.rs

```diff
@@ -0,0 +1,108 @@
+use anyhow::Result;
+use geo::{Euclidean, Length};
+use geojson::GeoJson;
+use serde::Deserialize;
+
+use crate::{
+    Kind, Speedwalk,
+    graph::{Edge, Graph},
+};
+
+#[derive(Deserialize)]
+pub struct NetworkFilter {
+    include: NetworkFilterType,
+    ignore_deadends: bool,
+}
+
+#[derive(Deserialize)]
+pub enum NetworkFilterType {
+    Everything,
+    OnlyExplicitFootways,
+    RouteableNetwork,
+}
+
+const MINIMUM_DEADEND_LENGTH: f64 = 10.0;
+
+impl Speedwalk {
+    pub fn filter_network(&self, filter: &NetworkFilter, graph: &Graph, edge: &Edge) -> bool {
+        let way = &self.derived_ways[&edge.osm_way];
+
+        match filter.include {
+            NetworkFilterType::Everything => {}
+            NetworkFilterType::OnlyExplicitFootways => {
+                if !matches!(way.kind, Kind::Sidewalk | Kind::Crossing | Kind::Other) {
+                    return false;
+                }
+            }
+            NetworkFilterType::RouteableNetwork => {
+                let include = match way.kind {
+                    // Use the separate footways to route
+                    // TODO Even if it's only separate on one side, but tagged for the other?
+                    Kind::RoadWithSeparate => false,
+                    Kind::RoadWithTags => true,
```

Comment:
So if a road has `sidewalk=left,right,both`, we include its centerline. If the user runs the automatic sidewalk generation, most/all of these should go away and become separate sidewalks